addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1041,6 +1041,122 @@ const ColorpickerUserValueWidget = SelectUserValueWidget.extend({
+        this.editImageButton = document.createElement('we-button');
+        this.editImageButton.classList.add('o_we_pick_image');
+        this.editImageButton.append(this.editBgTextEl);
+        this.editImageButton.append(editIconEl);
+        this.editImageButton.addEventListener('click', (ev) => this._onEditImage(ev));
+
+        this.removeImageButton = document.createElement('we-button');
+        this.removeImageButton.classList.add('fa', 'fa-fw', 'fa-times');
+        this.removeImageButton.title = _t("Remove image");
+        this.removeImageButton.addEventListener('click', (ev) => this._onRemoveImage(ev));
+
+        return this.containerEl.append(_buildRowElement('', {
+            childNodes: [this.editImageButton, this.removeImageButton],
+        }));
+    },
+    /**
+     * @override
+     */
+    getMethodsParams: function (methodName) {
+        return _.extend({isVideo: this.isVideo}, this._super(...arguments));
+    },
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    updateUI: async function () {
+        await this._super(...arguments);
+        this.removeImageButton.classList.toggle('d-none', !this.isActive());
+        this.editBgTextEl.innerText = this.isActive() ? this._getDisplayName() : this.defaultText;
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Computes the name to display in the edit button.
+     *
+     * @private
+     */
+    _getDisplayName: function () {
+        if (this.isVideo) {
+            return this._value;
+        }
+        return this._value.substring(this._value.lastIndexOf('/') + 1).split(/[?#]/)[0];
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Called when the edit image button is clicked.
+     *
+     * @private
+     */
+    _onEditImage: function (ev) {
+        // Need a dummy element for the media dialog to modify.
+        const dummyEl = document.createElement(this.isVideo ? 'iframe' : 'img');
+        dummyEl.src = this._value;
+        if (this.isVideo) {
+            // Allows the mediaDialog to select the video tab immediately.
+            dummyEl.classList.add('media_iframe_video');
+        }
+        const mediaDialog = new weWidgets.MediaDialog(this, {
+            noIcons: true,
+            noDocuments: true,
+            noVideos: !this.allowVideos,
+            isForBgVideo: true,
+        }, dummyEl).open();
+        mediaDialog.on('save', this, data => {
+            if (data.bgVideoSrc) {
+                this.setValue(data.bgVideoSrc);
+                this.isVideo = true;
```

Comment:
Mmh... in fact I think it is working in all normal cases but technically setValue is called from the outside when an option notifies the current state of the $target being edited... and in that case, technically, isVideo should be updated too. Don't know though :/ I suppose we can leave it this way

Reply:
This change is what allowed me to not have to externalize the getVideoURLData method to use it here. I could roll that back and reparse the value in setValue to check whether it's a video URL though, this way it could also be set from outside without side effects.

Reply:
Yes, I understood, let's leave it this way for now.